ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -256,9 +257,9 @@ public CompletableFuture<NamedBlobRecord> get(String accountName, String contain
         new GetTransactionStateTracker(remoteDatacenters, localDatacenter);
     return executeTransactionAsync(accountName, containerName, true, (accountId, containerId, connection) -> {
       long startTime = System.currentTimeMillis();
-      NamedBlobRecord record, recordV2;
+      NamedBlobRecord record, recordV2 = null;
       record = run_get(accountName, containerName, blobName, option, accountId, containerId, connection);
-      if (config.dbTransition) {
+      if (config.dbTransition || config.dbNewtable) {
```

Comment:
There are two problems with this code.
1. even when reading from new table, we are still throw exception for old table db operation and catching exception for new table operations.
2. reading from new table is a concept that is valid only when we are in the middle of transition. So it's not necessary to put these two boolean statements together.

I would recommend doing something like this
```
interface GetRecord {
    NamedBlobRecord do(String accountName, String containerName, String blobName, GetOption option, short accountId, short containerId, Connection connection) throws Exception;
}

NamedBlobRecord primary, secondary;
GetRecord primaryGetter = this.run_get, secondaryGetter = this.run_get_v2;
if (config.dbNewTable) {
    primaryGetter = this.run_get_v2;
    secondaryGetter = this.run_get;
}

primary = primaryGetter.do(xxxx);
if (config.dbTransition) {
    // do secondary
}
return primary;
```

Reply:
Those two problems are addressed in 7826a9b.
Thank you for the interface suggestion. If we go with that interface way, then we need to create multiples interfaces (for Get, List, Put, Delete). And I prefer to code the logic like below to be more explicit:

```
      try {
        result = operation_on_old_table(...);
      } catch (Exception eOld) {
        if (!config.dbNewtable) {
          throw eOld;
        }
      }

      if (config.dbTransition) {
        try {
          resultV2 = operation_on_new_table(...);
          ....
        } catch (Exception e) {
          if (config.dbNewtable) {
            throw e;
          }
          ....
        }
      }
     ...
      return config.dbNewtable? resultV2: result;
```